rein/lib/rating.py

```diff
@@ -100,9 +103,77 @@ def add_rating(rein, user, testnet, rating, user_msin, job_id, rated_by_msin, co
     store = True
     document = None
     if update_rating:
-    	document = sign_and_store_document(rein, 'rating', document_text, user.daddr, user.dkey, store, update_rating.doc_hash)
+    	flash(u"Ratings cannot currently be updated.")
 
     else:
     	document = sign_and_store_document(rein, 'rating', document_text, user.daddr, user.dkey, store)
     
-    return (document and store)
\ No newline at end of file
+    return (document and store)
+
+def get_average_user_rating(log, url, user, rein, msin):
+    """Gets the average rating a user (identified by his msin) has received
+    along with the number of ratings he has received"""
+
+    sel_url = "{0}query?owner={1}&delegate={2}&query=get_user_ratings&testnet={3}&msin={4}"
+    data = safe_get(log, sel_url.format(url, user.maddr, user.daddr, rein.testnet, msin))
+    data = data['get_user_ratings']
+
+    # If there was a server-side error, return None
+    if 'error' in data:
+        return None
+
+    # Create a list of all the ratings the user has received
+    rating_values = []
+    for rating_data in data:
+        rating = document_to_dict(rating_data['value'])
+        try:
+            rating_value = int(rating['Rating'])
+            rating_values.append(rating_value)
+
+        except:
+            pass
+
+    # Return None if the user has not been rated
+    if len(rating_values) == 0:
+        return None
+
+    average_rating = float(sum(rating_values)) / float(len(rating_values))
+    return (average_rating, len(rating_values))
+
+def get_averave_user_rating_display(log, url, user, rein, msin, cli=False):
+    """Returns a user's average rating in html format (as a link to the user's ratings page)."""
+
+    rating = get_average_user_rating(log, url, user, rein, msin)
+    if not rating:
+        return 'Not yet rated'
+
+    if not cli:
+        return '<a href="/ratings/{}">{} <i class="fa fa-star-o"></i> ({} ratings)</a>'.format(msin, rating[0], rating[1])
+
+    return '{} Stars ({} ratings)'.format(rating[0], rating[1]) 
```

Comment:
Is there an html code for star? We can remove the word ratings as well since I think a number in parens is usually clear enough.